types/sinon/index.d.ts

```diff
@@ -1685,12 +1685,16 @@ declare namespace Sinon {
         ): SinonStubbedInstance<TType>;
 
         /**
-         * Defines a property on the given object which will be torn down when
-         * the sandbox is restored
+         * Defines a property on the given object which will be torn down when the sandbox is restored.
+         * Attempts to define an already existing property will cause an exception.
+         * Attempts to pass undefined value will cause an exception.
+         * @param obj
+         * @param key
+         * @param value
          */
-        define(
-            obj: object,
-            key: PropertyKey,
+        define<T>(
```

Comment:
```suggestion
        define<T extends object = object>(
```

Can you add a test showing that `sb.define("notAnObject", "someKey", 100)` errors? The PR as it stands now removes that error.